### PR TITLE
API test for deleteShare

### PR DESCRIPTION
**Pull request by paurakhsharma**
## Description
This PR adds more API tests for deleting shares.

## Related Issue
#31531 

## How Has This Been Tested?
Locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Acceptance test

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: tests/acceptance/features/apiSharing-v1/deleteShare.feature [hunk 1]
```diff
@@ -24,17 +24,19 @@ Feature: sharing
...
 		And user "user1" has been created
 		And user "user0" has created a folder "/common"
 		And user "user0" has created a folder "/common/sub"
-		And user "user0" has shared file "/common/sub" with user "user1"
-		And user "user0" has deleted folder "/common"
-		When user "user0" empties the trashbin using the API
-		Then as "user1" the folder "/sub" should not exist
+		And user "user0" has shared folder "/common/sub" with user "user1"
+		When user "user0" deletes folder "/common" using the API
+		Then the OCS status code should be "100"
```
Comment (phil-davis) [inline tests/acceptance/features/apiSharing-v1/deleteShare.feature]: Something went wrong here in the CI

File: tests/acceptance/features/apiSharing-v1/deleteShare.feature [hunk 2]
```diff
@@ -98,4 +100,43 @@ Feature: sharing
 		And user "user1" has stored etag of element "/"
 		When user "user1" deletes the last share using the API
 		Then the etag of element "/" of user "user1" should have changed
-		And the etag of element "/PARENT" of user "user0" should not have changed
\ No newline at end of file
+		And the etag of element "/PARENT" of user "user0" should not have changed
+
+	Scenario: sharee of a read-only share folder tries to delete the shared folder
```
Comment (phil-davis) [inline tests/acceptance/features/apiSharing-v1/deleteShare.feature]: s/share folder/shared folder/

File: tests/acceptance/features/apiSharing-v1/deleteShare.feature [hunk 3]
```diff
@@ -98,4 +100,43 @@ Feature: sharing
...
+		Given user "user0" has been created
+		And user "user1" has been created
+		And user "user0" has created a folder "/shared"
+		And user "user0" has moved file "/textfile0.txt" to "/shared/shared_file.txt"
+		And user "user0" has sent HTTP method "POST" to API endpoint "/apps/files_sharing/api/v1/shares" with body
+			| path        | shared |
+			| shareWith   | user1  |
+			| shareType   | 0      |
+			| permissions | 1      |
+		When user "user1" deletes file "/shared/shared_file.txt" using the API
```
Comment (phil-davis) [inline tests/acceptance/features/apiSharing-v1/deleteShare.feature]: After this step, can we check OCS/HTTP status? Should be some fail code?
Same in the following scenarios...

File: tests/acceptance/features/apiSharing-v1/deleteShare.feature [hunk 4]
```diff
@@ -98,4 +100,43 @@ Feature: sharing
...
+		Then as "user1" the file "/shared/shared_file.txt" should exist
+
+	Scenario: sharee of a upload-only share folder tries to delete the shared folder
```
Comment (phil-davis) [inline tests/acceptance/features/apiSharing-v1/deleteShare.feature]: s/share folder/shared folder/

"tries to delete a file in the shared folder"

File: tests/acceptance/features/apiSharing-v1/deleteShare.feature [hunk 5]
```diff
@@ -98,4 +100,43 @@ Feature: sharing
...
+		And user "user0" has moved file "/textfile0.txt" to "/shared/shared_file.txt"
+		And user "user0" has sent HTTP method "POST" to API endpoint "/apps/files_sharing/api/v1/shares" with body
+			| path        | shared |
+			| shareWith   | user1  |
+			| shareType   | 0      |
+			| permissions | 4      |
+		When user "user1" deletes file "/shared/shared_file.txt" using the API
+		Then as "user0" the file "/shared/shared_file.txt" should exist
+
+	Scenario: sharee of a upload-only share folder tries to delete his share in the folder
```
Comment (phil-davis) [inline tests/acceptance/features/apiSharing-v1/deleteShare.feature]: "sharee of a upload-only shared folder tries to delete a file that he uploaded in the folder"

**Review by phil-davis — COMMENTED** [review]
Few little comments...

**Review by phil-davis — APPROVED** [review]

File: tests/acceptance/features/apiSharing-v1/deleteShare.feature [hunk 6]
```diff
@@ -24,16 +24,17 @@ Feature: sharing
...
 		And user "user1" has been created
 		And user "user0" has created a folder "/common"
 		And user "user0" has created a folder "/common/sub"
-		And user "user0" has shared file "/common/sub" with user "user1"
-		And user "user0" has deleted folder "/common"
-		When user "user0" empties the trashbin using the API
+		And user "user0" has shared folder "/common/sub" with user "user1"
+		When user "user0" deletes folder "/common" using the API
+		Then the HTTP status code should be "204"
 		Then as "user1" the folder "/sub" should not exist
```
Comment (phil-davis) [inline tests/acceptance/features/apiSharing-v1/deleteShare.feature]: s/Then/And

File: tests/acceptance/features/apiSharing-v1/deleteShare.feature (hunk 3 above)
Comment (phil-davis) [reply tests/acceptance/features/apiSharing-v1/deleteShare.feature]: done

File: tests/acceptance/features/apiSharing-v1/deleteShare.feature [hunk 7]
```diff
@@ -97,5 +100,48 @@ Feature: sharing
...
+		And user "user0" has sent HTTP method "POST" to API endpoint "/apps/files_sharing/api/v1/shares" with body
+			| path        | shared |
+			| shareWith   | user1  |
+			| shareType   | 0      |
+			| permissions | 4      |
+		When user "user1" deletes file "/shared/shared_file.txt" using the API
+		Then the HTTP status code should be "403"
+		And as "user0" the file "/shared/shared_file.txt" should exist
+
+	Scenario: sharee of a upload-only share folder tries to delete his share in the folder
```
Comment (phil-davis) [inline tests/acceptance/features/apiSharing-v1/deleteShare.feature]: tries to delete his file in the folder

**Comment by phil-davis** [comment]
All comments have been addressed. Ready for merge when CI  is finished.

**Comment by paurakhsharma** [comment]
Backport on #31574

**Pull request merged**